### ISSUE TITLE
EPAM_DX docs: tt-xla — xr.set_device_type('TT') silently fails on venv-only host; canonical contract is PJRT_DEVICE=TT

### ISSUE THREAD

**Issue by shuanat** [tenstorrent/tt-metal#43878]
### Summary

The EPAM Round 4 internal assignment checklist sets up a Python venv in one step and then references `xr.set_device_type('TT')` in a later "framework integration" step. This call is meaningful only inside the `tt-xla-slim` container — the public `torch_xla` from PyPI does not register the `'TT'` device type. Following the checklist linearly therefore produces a silent failure between the venv-install step and the framework-integration step.

Additionally, the canonical TT-XLA contract (verified end-to-end inside the container) is to set `PJRT_DEVICE=TT` as an env var and **not** to call `xr.set_device_type("TT")`. The checklist still uses the latter pattern.

### Scope

Vendor `tt-xla` docs and the `tt-xla-slim` published image both use `PJRT_DEVICE=TT`. The `xr.set_device_type('TT')` + venv-only sequence exists only inside the EPAM Round 4 program's internal assignment checklist and DX rubric. Filed here per the program's Journey-Based Testing guide, which routes EPAM_DX feedback to `tenstorrent/tt-metal`. Maintainers can treat this as informational; the corrective action is on the EPAM side.

### Classification

- Type: doc gap / blocker (as-written sequence is internally inconsistent)
- Severity: blocking — users hit a silent failure at the first TT-touching call.
- Priority: high
- Observed: 2026-05-07

### Vendor claim / documented path

- Claim: After installing Forge into a venv, `xr.set_device_type('TT')` registers the TT device for `torch_xla`.
- Source: developer-journey checklist framework-integration step; DX rubric "Framework Integration Check" row.
- Command was copied verbatim from the docs.

### Repro — failure on the venv-only path the checklist sets up

```bash
python -m venv /tmp/v && source /tmp/v/bin/activate
pip install torch torch-xla
python -c "import torch_xla.runtime as xr; xr.set_device_type('TT'); \
           import torch_xla.core.xla_model as xm; print(xm.xla_device())"
# → silent failure: the 'TT' device type is not registered by the public torch_xla wheel.
```

### Reference — what actually works (inside the container, with the correct API)

```bash
# JAX path:
podman run --rm \
  --device=/dev/tenstorrent \
  -v /dev/hugepages-1G:/dev/hugepages-1G \
  ghcr.io/tenstorrent/tt-xla-slim:latest \
  python -c "import jax; print(jax.devices('tt'))"
# → [TTDevice(id=0..7)]

# torch_xla path:
podman run --rm \
  --device=/dev/tenstorrent \
  -v /dev/hugepages-1G:/dev/hugepages-1G \
  --env=PJRT_DEVICE=TT \
  ghcr.io/tenstorrent/tt-xla-slim:latest \
  python -c "import torch_xla.core.xla_model as xm; print(xm.xla_device())"
# → xla:0
```

Note the working torch_xla invocation uses **`PJRT_DEVICE=TT`** as an env var; it does **not** call `xr.set_device_type("TT")`.

### Expected

The documented sequence is internally consistent: either the framework-integration step runs **inside** the Forge container (and the previous step makes that explicit), or the venv install path actually delivers a `torch_xla` build that registers the TT device.

### Actual

The two steps are stitched together implicitly. Worse, the call used in the checklist (`xr.set_device_type('TT')`) is not the canonical TT-XLA contract; the supported entry point is the `PJRT_DEVICE=TT` env var.

### Evidence

Verified on a TT lab host (n300/wormhole, 8 devices), 2026-05-07:

- `import jax; jax.devices('tt')` inside `tt-xla-slim:latest` returns `[TTDevice(id=0)..TTDevice(id=7)]`.
- `import torch_xla.core.xla_model as xm; xm.xla_device()` inside `tt-xla-slim:latest` with `PJRT_DEVICE=TT` returns `xla:0`. Subsequent `torch.zeros(2,3, device=device)` succeeds and roundtrips back to CPU as `torch.Size([2, 3])`.
- The same `xm.xla_device()` call from a fresh venv with `pip install torch torch-xla` (no container) does not produce a TT device. The public `torch_xla` wheel has no PJRT TT plugin.
- `xr.set_device_type('TT')` is not part of the working invocation in either case (container or venv); the documented call is therefore not the canonical contract for the current `tt-xla` stack.

### Impact

Participants who follow the checklist linearly hit a silent failure between the install and framework-integration steps. Combined with the missing pip-flow for `tt-forge` (separate sibling report), the documented bring-up sequence is unrunnable as-written.

### Suggested fix

Two compatible parts:

1. Make the framework-integration step say explicitly "run inside the Forge container from Step N" (and provide the container invocation, or equivalent active-venv path).
2. Replace `xr.set_device_type('TT')` with `PJRT_DEVICE=TT` (env var) per the current TT-XLA contract.

### Tracking

`EPAM_DX` — filed as part of the EPAM Round 4 Tenstorrent developer-journey program. If the `EPAM_DX` label cannot be applied here, this mention serves the program-side text search per the Journey-Based Testing guide.
